library/cartridge_get_package_info.py

```diff
@@ -140,11 +146,11 @@ def get_package_info(params):
     elif package_type == 'deb':
         package_info = get_deb_info(package_path)
     elif package_type == 'tgz':
-        package_info = get_tgz_info(package_path, app_name)
+        package_info = get_tgz_info(package_path)
     else:
         return helpers.ModuleRes(failed=True, msg='Unknown package type: %s' % package_type)
 
-    if app_name and package_info['name'] != app_name:
+    if package_type != 'tgz' and app_name and package_info['name'] != app_name:
```

Comment:
app_name is required, so:
```suggestion
    if package_type != 'tgz' and package_info['name'] != app_name:
```